azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1729,6 +1727,28 @@ private void cleanExecutionLogs() {
     }
   }
 
+  /**
+   * Calls executor to dispatch the flow, update db to assign the executor and
+   * in-memory state of executableFlow
+   */
+  private void dispatch(ExecutionReference reference, ExecutableFlow exflow,
+    Executor choosenExecutor) throws ExecutorManagerException {
+    exflow.setUpdateTime(System.currentTimeMillis());
```

Comment:
no we may fail to call executor and our priority queue logic depends on update time. Moreover, executor server will update updatetime once it is successfully dispatched.